website/src/components/TutorialComponents/runZ3DuoWeb.ts

```diff
@@ -1,23 +1,37 @@
 import loadZ3 from './loadZ3';
 
+import { Z3_error_code, } from 'z3-solver';
+
 export default async function runZ3DuoWeb(user_input: string, secret_input: string): Promise<string> {
 
     // init z3
     const z3p = loadZ3();
 
-    const { Context } = await z3p;
+    const { Context, Z3: Z3Core } = await z3p;
     let Z3 = Context('main');
 
     let output = '';
     let error = '';
     let outputObj;
 
+    const throwIfError = (ctxPtr) => {
+        const errorCode = Z3Core.get_error_code(ctxPtr);
+        console.log(ctxPtr)
+        console.log({errorCode})
+        if (Z3Core.get_error_code(ctxPtr) !== Z3_error_code.Z3_OK) {
+            throw new Error(Z3Core.get_error_msg(ctxPtr, Z3Core.get_error_code(ctxPtr)));
+        }
+    }
+
     try {
         const s1 = new Z3.Solver();
         const s2 = new Z3.Solver();
         s1.fromString(user_input);
         s2.fromString(secret_input);
 
+        throwIfError(s1.ptr);
```

Comment:
ptr is the pointer to the context or solver?

Reply:
I think you want s1.ctx.ptr

Reply:
export interface Solver<Name extends string = 'main'> {
  /** @hidden */
  readonly __typename: 'Solver';

  readonly ctx: Context<Name>;
  readonly ptr: Z3_solver;

Reply:
export interface Context<Name extends string = 'main'> {
  /** @hidden */
  readonly ptr: Z3_context;
  /**

Reply:
the error check code takes a context pointer (Z3_context)